### ISSUE TITLE
[Flatpak] Browser Sources stopped working after latest update

### ISSUE THREAD

**Issue by VortexAcherontic** [obsproject/obs-studio#10385]
### Operating System Info

Other

### Other OS

openSUSE Aeon

### OBS Studio Version

30.1.0

### OBS Studio Log URL

https://obsproject.com/logs/pCOkzB18wC0mPCoM

### Expected Behavior

- Browser sources do work

### Current Behavior

- Browser sources do not work

### Steps to Reproduce

1. Install OBS Studio 30.1.0
2. Add browser source
3. Note they do not work

### Anything else we should know?

Browser sources seem empty. Even static HTML sites just keep blank. From a logs perspective they seem to work fine while they don't.
But also I noticed browser sources which usually do test if the browser source is active to display panels on stream do not do so either. Like soundalerts for example.
With the previous version 30.0.2 everything was working fine with no other changes to my system except regular updates.
But since I use the flatpak the system should not be interfering with he OBS Studio install.

After manually rolling back to the previous release using:

```
flatpak update --commit=e641d66f8f509ff18c1e82bfd6e32e0e0021e56cb01d1471b00f56d82ea8bfa1 com.obsproject.Studio
```

Browser sources started working again using the same system and app configuration

**Comment by RytoEX** [comment]
So far, this seems exclusive to OpenSUSE. Works fine on Ubuntu. It would be good to know if this is also broken in any of the previous betas, but you would probably have to manually install them from our CI builds.

**Comment by VortexAcherontic** [comment]
> So far, this seems exclusive to OpenSUSE. Works fine on Ubuntu. It would be helpful to know if this is also broken in any of the previous betas, but you would probably have to manually install them from our CI builds.

Just to be sure, you've ran the flatpak OBS Studio as well? Asking because as of now I thought the the main idea of flatpak is to be independent of the host system due to the flatpak runtimes. So that an app should run identically on any distribution.

**Comment by VortexAcherontic** [comment]
I just noticed another thing. OBS Studio runs with Flatpak Runtime: runtime/org.kde.Platform/x86_64/6.6 while 30.0.2 runs with Flatpak Runtime: runtime/org.kde.Platform/x86_64/6.5 Maybe there is something different in the runtimes.

**Comment by VortexAcherontic** [comment]
I installed the *.dep 30.1.0 inside a Ubuntu distrobox container running on my host system and browser sources are working as expected. Maybe something is just broken with the flatpak?

**Comment by RytoEX** [comment]
> > So far, this seems exclusive to OpenSUSE. Works fine on Ubuntu. It would be helpful to know if this is also broken in any of the previous betas, but you would probably have to manually install them from our CI builds.
> 
> Just to be sure, you've ran the flatpak OBS Studio as well? Asking because as of now I thought the the main idea of flatpak is to be independent of the host system due to the flatpak runtimes. So that an app should run identically on any distribution.

I, myself, did not. I primarily work on Windows. Someone else tested on OpenSUSE for me and confirmed that it was having the described issue. Yet another person tested the Flatpak on Ubuntu and said it worked fine.

**Comment by VortexAcherontic** [comment]
Downgraded (30.0.2) OBS Studio flatpak https://obsproject.com/logs/MIOoOTbZBJ3JaOE2

**Comment by VortexAcherontic** [comment]
To keep track of the findings on Discord:

- It turned out to be an issue with https sources as it seems on openSUSE (Leap, Tumbleweed, Aeon and Kalpa) it is unable to verify the SSL certificate for some unknown reason. This applies to Browser Docks as well as Browser sources inside a Scene.
- Locally building and running OBS Studio 30.1.0 with runtime 6.5 does work with browser sources.
- Locally building 30.1.0 with runtime 6.6 does not work
- As of now the issue is exclusive to openSUSE, tested Ubuntu myself in a VM using the flatpak and things worked as expected.

Maybe I should pass this issue to the runtime instead of OBS Studio then.

Thank you Tarulia, Cookie and feaneron for assisting here ❤️

**Comment by VortexAcherontic** [comment]
Submitted bug report to KDE: https://bugs.kde.org/show_bug.cgi?id=483746

**Comment by VortexAcherontic** [comment]
Bug also reported to openSUSE: https://bugzilla.opensuse.org/show_bug.cgi?id=1221557

**Comment by VortexAcherontic** [comment]
To keep track of further findings:
Running OBS Studio 30.1.0 natively on openSUSE using Qt 6.6.2 (Which is what the Runtime 6.6 ships as well) works fine.
Also both the Tumbleweed container and the runtime 6.6 make use of libssl3.

**Comment by VortexAcherontic** [comment]
Guys it seems I am running in circles it seems. Testing other KDE apps which do also use runtime 6.6 like neochat do work fine with SSL. Would it be possible that the embedded chromium has issues with SSL under certain circumstances with runtime 6.6? Does anyone know how I might be able to test this?

Using the debug console of a dock isn't help full. It just say nothing at all except the error message but no logs in the dev tools.

**Comment by mendres82** [comment]
Do Flatpaks come with their own certificate store or are they using the ones that are on the "host system"? Didn't have the time to dig into yet (updated my system to Gnome 46 and fixing extension shenanigans here and there), but could it be possible that certificate verification fails because of missing root certificates?

**Comment by VortexAcherontic** [comment]
> Do Flatpaks come with their own certificate store or are they using the ones that are on the "host system"? Didn't have the time to dig into yet (updated my system to Gnome 46 and fixing extension shenanigans here and there), but could it be possible that certificate verification fails because of missing root certificates?

Other flat do work fine. Even KDE Apps making also use of the KDE Runtime 6.6 like neochat. As of now I was only able to reproduce this issue with OBS Studio and only with the embedded chromium. Other stuff like streaming and such did still work just fine.

**Comment by mendres82** [comment]
Hm, I see. Out of curiosity I looked into the runtime flat and it comes with all the certificates needed. So this shouldn't be a problem, just like you said.

**Comment by My2ndAngelic** [comment]
I tested this with Fedora KDE and Kubuntu, both of them works fine with the newest flatpak release. This problem occurs only with the newest release of OBS on openSUSE Tumbleweed afaik. 

Seems very strange tbh.

**Comment by VortexAcherontic** [comment]
> This problem occurs only with that release of OBS on openSUSE Tumbleweed afaik.

openSUSE Leap 15.5 is broken as well.

**Comment by Matsa59** [comment]
I still have the problem, as mention the author of this issue, the following command make browser source work again.

```
flatpak update --commit=e641d66f8f509ff18c1e82bfd6e32e0e0021e56cb01d1471b00f56d82ea8bfa1 com.obsproject.Studio
```

Thx mate

**Comment by mihawk90** [comment]
Just to keep it documented:
Today we received a support request for the same issue, which is interesting because it *wasn't* on SUSE, but on Alpine Linux.

The "workaround" via downgrade worked here too, we have not tested building with KDER 6.5 as of yet.

**Comment by tytan652** [comment]
I dug a little by downgrading `org.kde.Platform//6.6` before KF6 was added and it did not fix it.

So the issue mainly relies on the distro and Freedesktop SDK 23.08…

**Comment by mihawk90** [comment]
Got another report on Discord, this time on MX Linux... downgrading worked here as well.

**Comment by mihawk90** [comment]
Just had another report on Discord for Peppermint Linux, Debian based.

Whatever it is, it's spreading :/

**Comment by mrivero** [comment]
I haven't used OBS in a while (OpenSuse - Flatpak) and it's doing the same thing for my browser objects (or even the embedded twitch connect).... downgrading works, but is there a way to downgrade its NDI plugin that correspond to the 30.0.2 version as well? (I've tried a few older commit numbers for it but they are erroring)

**Comment by p0lfilt3r** [comment]
Thank you @MinimusMaximus that solved it for me.

After installing p11-kit-server on OpenSUSE Leap 15.6 and a reboot (!!!) it works!

**Comment by derrod** [comment]
That package replaces the `mozialla-nss-certs` package, perhaps that one has patches that made it incompatible with Flatpak on openSUSE?

**Comment by VortexAcherontic** [comment]
> That package replaces the `mozialla-nss-certs` package, perhaps that one has patches that made it incompatible with Flatpak on openSUSE?

Well looking up the Tumbleweed package there are quite a lot patches for various reasons 😅

https://build.opensuse.org/projects/openSUSE:Factory/packages/mozilla-nss/files/mozilla-nss.spec?expand=1

**Comment by derrod** [comment]
Interesting, so perhaps an issue with openSUSE's flatpak image missing that dependency.

**Comment by mrivero** [comment]
> I solved a SSL error in another flatpak application recently by installing [p11-kit-server](https://software.opensuse.org/package/p11-kit-server?search_term=p11-kit-server) package. After installing the package, log out and log in or restart the system.
@MinimusMaximus confirmed this just worked for me with the OBS browser error. Great finding, thank you!

**Comment by Anthirian** [comment]
This is still an issue as of version 30.1.2. I have read solutions online that suggest disabling hardware acceleration through the settings, but the option seems to be absent too.

**Comment by VortexAcherontic** [comment]
> This is still an issue as of version 30.1.2 on OpenSUSE Tumbleweed. I have read solutions online that suggest disabling hardware acceleration through the settings, but the option seems to be absent too.

You may want to install p11-kit-server on TW. Didn’t tried myself but others confirmed it works.

Anyhow I would also love if ppl could rather urge the KDE devs to fix their runtime instead.

**Comment by mihawk90** [comment]
I didn't note it here, but I did update them regarding our findings on their bug tracker about a week and a half ago.

**Comment by VortexAcherontic** [comment]
Too keep this updated. For openSUSE Tumbleweed the issue is soon to be fixed as p11-kit-server was added as weak dependency for their flatpak package via this request: https://build.opensuse.org/request/show/1192619 in the coming week this issue should solve it self for Tumbleweed users.

**Comment by mihawk90** [comment]
KDE also closed their bug:
https://bugs.kde.org/show_bug.cgi?id=483746#c10

**Comment by Sturmlocke86** [comment]
I just had the same issue on Leap 15.6 with Gnome and through a bit of troubleshooting I was able to fix this.

- It's this issue: https://github.com/obsproject/obs-studio/issues/8880
- ERR_CERT_AUTHORITY_INVALID is displayed when using the Youtube Live Control Panel and Chat dock or when trying to add custom sources etc.
- The issue has been discussed here: https://www.reddit.com/r/openSUSE/comments/1d56u29/obs_issue_ssl_cert_error/
- In short, as mentioned in this bug report (#10385), the solution is to install "p11-kit-server" (it's two clicks via Yast -> Software): https://software.opensuse.org/package/p11-kit-server?search_term=p11-kit-server
- The bug was fixed on Tumbleweed but wasn't fixed for me on Leap 15.6 (upgraded from 15.5 to 15.6 Leap via USB stick): https://build.opensuse.org/request/show/1192619

**Verdict:** I just installed p11-kit-server via Yast -> Software five minutes ago and tested OBS with the docks mentioned further above, and it worked like a charm. Both docks are connecting to Youtube just fine now. Leaving this as feedback with the plea of performing a backport for Leap users. Thanks.
